webapp-django/crashstats/supersearch/models.py

```diff
@@ -1,101 +1,34 @@
+import functools
 import json
 
+from django.core.cache import cache
+
 from crashstats import scrubber
 from crashstats.crashstats import models
 
 
-SUPERSEARCH_HITS_WHITELIST = (
-    'additional_minidumps',
-    'addons',
-    'addons_checked',
-    'address',
-    'app_notes',
-    'build_id',
-    'client_crash_date',
-    'completeddatetime',
-    'cpu_info',
-    'cpu_name',
-    'crashedThread',
-    'crash_time',
-    'date',
-    'distributor',
-    'distributor_version',
-    'flash_version',
-    'hangid',
-    'hang_type',
-    'id',
-    'install_age',
-    'java_stack_trace',
-    'last_crash',
-    'platform',
-    'platform_version',
-    'plugin_filename',
-    'plugin_name',
-    'plugin_version',
-    'processor_notes',
-    'process_type',
-    'product',
-    'productid',
-    'reason',
-    'release_channel',
-    'ReleaseChannel',
-    'signature',
-    'startedDateTime',
-    'success',
-    'topmost_filenames',
-    'truncated',
-    'uptime',
-    'user_comments',
-    'uuid',
-    'version',
-    'winsock_lsp',
-    'accessibility',
-    'adapter_device_id',
-    'adapter_vendor_id',
-    'android_board',
-    'android_brand',
-    'android_cpu_abi',
-    'android_cpu_abi2',
-    'android_device',
-    'android_display',
-    'android_fingerprint',
-    'android_hardware',
-    'android_manufacturer',
-    'android_model',
-    'android_version',
-    'async_shutdown_timeout',
-    'available_page_file',
-    'available_physical_memory',
-    'available_virtual_memory',
-    'b2g_os_version',
-    'bios_manufacturer',
-    'cpu_usage_flash_process1',
-    'cpu_usage_flash_process2',
-    'dom_ipc_enabled',
-    'em_check_compatibility',
-    'frame_poison_base',
-    'frame_poison_size',
-    'is_garbage_collecting',
-    'min_arm_version',
-    'number_of_processors',
-    'oom_allocation_size',
-    'plugin_cpu_usage',
-    'plugin_hang',
-    'plugin_hang_ui_duration',
-    'startup_time',
-    'system_memory_use_percentage',
-    'theme',
-    'throttleable',
-    'total_virtual_memory',
-    'vendor',
-    'additional_minidumps',
-    'throttle_rate',
-    'useragent_locale',
-    # deliberately not including:
-    #    email
-    #    url
-    #    exploitability
-)
+def get_api_whitelist(*args, **kwargs):
+
+    def get_from_es(unredacted=False):
+        fields = cache.get('api_supersearch_fields')
+        if fields is None:
+            all = SuperSearchFields().get()
+            fields = []
+            for meta in all.itervalues():
+                if (
+                    meta['is_returned'] and (
+                        unredacted or
+                        not meta['permissions_needed']
+                    )
+                ):
+                    if meta['name'] not in fields:
```

Comment:
just add this to the parent if statement. It's weird that it's not part of it. Is there a reason for this?